nextcloud-server.yaml

```diff
@@ -0,0 +1,388 @@
+      - php-${{vars.php-version}}-config
+      - php-${{vars.php-version}}-apache
+      - php-${{vars.php-version}}-bcmath
+      - php-${{vars.php-version}}-exif
+      - php-${{vars.php-version}}-ftp
+      - php-${{vars.php-version}}-fpm
+      - php-${{vars.php-version}}-gd
+      - php-${{vars.php-version}}-gmp
+      - php-${{vars.php-version}}-intl
+      - php-${{vars.php-version}}-apcu
+      - php-${{vars.php-version}}-ldap
+      - php-${{vars.php-version}}-igbinary
+      - php-${{vars.php-version}}-imagick
+      - php-${{vars.php-version}}-memcached
+      - php-${{vars.php-version}}-sodium
+      - php-${{vars.php-version}}-opcache
+      - php-${{vars.php-version}}-pcntl
+      - php-${{vars.php-version}}-pdo
+      - php-${{vars.php-version}}-pdo_mysql
+      - php-${{vars.php-version}}-mysqlnd
+      - php-${{vars.php-version}}-pdo_pgsql
+      - php-${{vars.php-version}}-sysvsem
+      - php-${{vars.php-version}}-zip
+      - php-${{vars.php-version}}-redis
+      - php-${{vars.php-version}}-pdo_sqlite
+      - php-${{vars.php-version}}-iconv
+      - php-${{vars.php-version}}-ctype
+      - php-${{vars.php-version}}-dom
+      - php-${{vars.php-version}}-curl
+      - php-${{vars.php-version}}-xml
+      - php-${{vars.php-version}}-simplexml
+      - php-${{vars.php-version}}-fileinfo
+      - php-${{vars.php-version}}-mbstring
+      - php-${{vars.php-version}}-curl
+      - php-${{vars.php-version}}-openssl
+      - php-${{vars.php-version}}-posix
+      - php-${{vars.php-version}}-xmlreader
+      - php-${{vars.php-version}}-xmlwriter
+      - rsync
+      - ${{package.name}}-apache2-config
+  scriptlets:
+    post-install: |
+      #!/bin/sh
+      # As wolfi-baselayout does provide /var/spool/cron already, and we can not create this
+      # directory in the package, we need to create the cron file in the post-install scriptlet.
+      echo '*/5 * * * * php -f /var/www/html/cron.php' > /var/spool/cron/crontabs/www-data
+
+vars:
+  php-version: 8.1
+
+# Create a new major-version variable that contains only the major version
+# to use in the `nextcloud/docker` repo to find out the correct folder for the image.
+# e.g. 31.0.4 will create a new var major-version=31
+var-transforms:
+  - from: ${{package.version}}
+    match: ^(\d+).*
+    replace: $1
+    to: major-version
+
+environment:
+  contents:
+    packages:
+      - apache2-compat
+      - bash
+      - build-base
+      - busybox
+      - ca-certificates-bundle
+      - composer
+      - curl
+      - git
+      - nodejs
+      - npm
+      - php-${{vars.php-version}}
+  environment:
+    APACHE_BODY_LIMIT: "1073741824"
+    PHP_MEMORY_LIMIT: "512M"
+    PHP_UPLOAD_LIMIT: "512M"
+    PHP_OPCACHE_MEMORY_CONSUMPTION: "128"
+    PHP_INI_DIR: "etc/php"
+
+pipeline:
+  - runs: mkdir -p ${{targets.destdir}}/usr/src/nextcloud
+
+  - uses: fetch
+    with:
+      uri: https://download.nextcloud.com/server/releases/nextcloud-${{package.version}}.tar.bz2
```

Comment:
Curious why this is using fetch instead of git clone from https://github.com/nextcloud/server ?

Reply:
IIRC they release their source code by omitting other lots of unnecessary stuff (that we don't need), and since PHP is not a compiled lang. The tarball should be exactly same in the GitHub, so this is OK.

Reply:
done!